hal/src/electron/modem/mdm_hal.h

```diff
@@ -30,6 +30,105 @@
 #include "system_tick_hal.h"
 #include "cellular_enums_hal.h"
 #include "mdm_debug.h"
+#include "timer_hal.h"
+
+class CellularRegistrationStatusMdm {
```

Comment:
This is one to one to `CellularRegistrationStatus` in `cellular_ncp_client.h`. Let's just move the class into a new common header under `shared` to be used universally across Gen 2 and Gen 3.

Reply:
Good idea. 👍  Added.